engine/src/main/java/org/terasology/network/BroadCastServer.java

```diff
@@ -0,0 +1,139 @@
+        byte[] buffer = DISCOVERY_REQUEST.getBytes(StandardCharsets.UTF_8);
+        DatagramPacket packet = new DatagramPacket(buffer, buffer.length);
+        getReceiveSocket().receive(packet);
+        logger.info("Discovery package received! -> " + packet.getAddress() + ":" + packet.getPort());
+
+        //Validate the data sent.
+        String data = new String(packet.getData()).trim();
+        if (data.equals(DISCOVERY_REQUEST)) { // validate command
+            // Send response
+            byte[] response = new byte[DISCOVERY_RESPONSE.length()];
+            DatagramPacket responsePacket = new DatagramPacket(response, response.length, packet.getAddress(), packet.getPort());
+            getReceiveSocket().send(responsePacket);
+            logger.info("Response sent to: " + packet.getAddress() + ":" + packet.getPort());
+        } else {
+            logger.info("Error, not valid package!" + packet.getAddress() + ":" + packet.getPort());
+        }
+    }
+
+    private DatagramSocket getReceiveSocket()  {
+        try {
+            if (receiveSocket == null) {
+                receiveSocket = new DatagramSocket(8002,
+                        InetAddress.getByName("0.0.0.0")); // 0.0.0.0 for listen to all ips
+                receiveSocket.setBroadcast(true);
+            }
+        } catch (Exception e) {
+            logger.error("Broadcast Exception Encountered" + e.getMessage());
+        }
+        return receiveSocket;
+    }
+
+    private DatagramSocket getSendSocket() {
+        try {
+            if (sendSocket == null) {
+                sendSocket = new DatagramSocket(8001, InetAddress.getLocalHost());
+                sendSocket.setBroadcast(true);
+            }
+        } catch (Exception e) {
+            logger.error("Broadcast Exception Encountered" + e.getMessage());
+        }
+        return sendSocket;
+    }
+
+    public void sendBroadCast() throws Exception {
+        // Discovery request command
+        byte[] data = DISCOVERY_REQUEST.getBytes(StandardCharsets.UTF_8);
+        DatagramPacket packet = new DatagramPacket(data, data.length, InetAddress.getByName("255.255.255.255"), 8002);
+        getSendSocket().send(packet);
+        logger.info("Discovery package sent!" + packet.getAddress() + ":" + packet.getPort());
+
+        // Discovery response command
+        byte[] response = new byte[DISCOVERY_RESPONSE.length()];
+        DatagramPacket responsePacket = new DatagramPacket(response, response.length);
+        getSendSocket().receive(responsePacket);
+        logger.info("Discovery response received!" + responsePacket.getAddress() + ":" + responsePacket.getPort());
+        String responseData = new String(responsePacket.getData(), StandardCharsets.UTF_8);
+        if (responseData.equals(DISCOVERY_RESPONSE)) { // Check valid command
+            logger.info("Found server!" + responsePacket.getAddress() + ":" + responsePacket.getPort());
+        }
+    }
+}
```

Comment:
More trivial code style: Empty newline at the bottom of every text file please

Reply:
Ok